--- a/buffer.c
+++ b/buffer.c
@@ -386,11 +386,35 @@
 bool open_sbuf( void )
   {
   isbinary_ = false; reset_unterminated_line();
-  sfp = tmpfile();
+  const char * tmpdir = getenv( "TMPDIR" );
+  if( !tmpdir )
+    {
+    if( access( "/tmp", F_OK ) == 0 ) tmpdir = "/tmp";
+    else tmpdir = "/data/storage/el2/base/cache";
+    }
+  char * name;
+  if( asprintf( &name, "%s/ed-XXXXXX", tmpdir ) < 0 )
+    {
+    show_strerror( 0, errno );
+    set_error_msg( "Cannot open temp file" );
+    return false;
+    }
+  int fd = mkstemp( name );
+  if( fd < 0 )
+    {
+    show_strerror( 0, errno );
+    set_error_msg( "Cannot open temp file" );
+    free( name );
+    return false;
+    }
+  sfp = fdopen( fd, "w+" );
+  unlink( name );
+  free( name );
   if( !sfp )
     {
     show_strerror( 0, errno );
     set_error_msg( "Cannot open temp file" );
+    close( fd );
     return false;
     }
   return true;
