--- a/src/google/protobuf/parse_context.h
+++ b/src/google/protobuf/parse_context.h
@@ -393,10 +393,10 @@ class PROTOBUF_EXPORT EpsCopyInputStream
     ptr += round_down_size;
     // TODO: b/496023113 - Remove hexagon once hexagon-clang handles it
     // properly.
-#if !defined(__APPLE__) && !defined(__hexagon__)
+#if !defined(__APPLE__) && !defined(__hexagon__) && !defined(__OHOS__)
     // Working around clang issue on Apple and Hexagon.
     PROTOBUF_ALWAYS_INLINE_CALL
-#endif  // !__APPLE__ && !__hexagon__
+#endif  // !__APPLE__ && !__hexagon__ && !__OHOS__
     sink.Flush(ptr);
     size -= round_down_size;
 
@@ -424,10 +424,10 @@ class PROTOBUF_EXPORT EpsCopyInputStream
       if (!peek_func(view)) return nullptr;
       // TODO: b/496023113 - Remove hexagon once hexagon-clang handles it
       // properly.
-#if !defined(__APPLE__) && !defined(__hexagon__)
+#if !defined(__APPLE__) && !defined(__hexagon__) && !defined(__OHOS__)
       // Working around clang issue on Apple and Hexagon.
       PROTOBUF_ALWAYS_INLINE_CALL
-#endif  // !__APPLE__ && !__hexagon__
+#endif  // !__APPLE__ && !__hexagon__ && !__OHOS__
       sink.Append(view);
 
       ptr += append_size;
--- a/src/google/protobuf/io/zero_copy_stream_impl.cc
+++ b/src/google/protobuf/io/zero_copy_stream_impl.cc
@@ -58,7 +58,7 @@ namespace {
 // POSIX_CLOSE_RESTART needs to be provided (and the value defines the behavior
 // of posix_close). We use it to decide if we can use posix_close.
 int robust_close(int fd) {
-#if defined(POSIX_CLOSE_RESTART)
+#if defined(POSIX_CLOSE_RESTART) && !defined(__OHOS__)
   return posix_close(fd, 0);
 #else
   return close(fd);
