diff --git a/src/bun_core/env.rs b/src/bun_core/env.rs
index 9f4c52ef8c..f0967c3bf5 100644
--- a/src/bun_core/env.rs
+++ b/src/bun_core/env.rs
@@ -41,10 +41,12 @@ pub const IS_LINUX: bool = cfg!(any(target_os = "linux", target_os = "android"))
 const IS_FREEBSD: bool = cfg!(target_os = "freebsd");
 /// kqueue-based event loop (macOS + FreeBSD share most of this path).
 pub const IS_KQUEUE: bool = IS_MAC || IS_FREEBSD;
-const IS_AARCH64: bool = cfg!(target_arch = "aarch64");
-const IS_X64: bool = cfg!(target_arch = "x86_64");
-pub const IS_MUSL: bool = cfg!(target_env = "musl");
+pub(crate) const IS_AARCH64: bool = cfg!(target_arch = "aarch64");
+pub(crate) const IS_X64: bool = cfg!(target_arch = "x86_64");
+pub const IS_MUSL: bool = cfg!(any(target_env = "musl", target_env = "ohos"));
 pub const IS_ANDROID: bool = cfg!(target_os = "android");
+pub const IS_OHOS: bool = cfg!(target_env = "ohos");
+pub const IS_GLIBC: bool = IS_LINUX && cfg!(target_env = "gnu");
 pub const ALLOW_ASSERT: bool = IS_DEBUG || IS_TEST || build_options::RELEASE_SAFE;
 pub const CI_ASSERT: bool =
     IS_DEBUG || IS_TEST || ENABLE_ASAN || (build_options::RELEASE_SAFE && IS_CANARY);
