diff --git a/Makefile.SH b/Makefile.SH
index c3b42bb7a3..55414bb987 100755
--- a/Makefile.SH
+++ b/Makefile.SH
@@ -182,6 +182,9 @@ EOT
 		;;
 	esac
 
+	# Force symbols in libperl.so to be globally visible for OHOS platform
+	shrpldflags="$shrpldflags -Wl,-z,global"
+
 	;;
 
 *)	pldlflags=''
diff --git a/perl_langinfo.h b/perl_langinfo.h
index 6798c1f272..3dda7f9cec 100644
--- a/perl_langinfo.h
+++ b/perl_langinfo.h
@@ -256,7 +256,7 @@ typedef int nl_item;    /* Substitute 'int' for emulated nl_langinfo() */
  *
  * Android is special in that it doesn't define any of these, but has the
  * corresponding category */
-#if defined(__ANDROID__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_ADDRESS)
+#if defined(__ANDROID__) || defined(__OHOS__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_ADDRESS)
 #  define _NL_ADDRESS_POSTAL_FMT   -58
 #  define _NL_ADDRESS_COUNTRY_NAME -59
 #  define _NL_ADDRESS_COUNTRY_POST -60
@@ -272,7 +272,7 @@ typedef int nl_item;    /* Substitute 'int' for emulated nl_langinfo() */
 
 #  define HAS_MISSING_LANGINFO_ITEM_
 #endif
-#if defined(__ANDROID__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_IDENTIFICATION)
+#if defined(__ANDROID__) || defined(__OHOS__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_IDENTIFICATION)
 #  define _NL_IDENTIFICATION_TITLE        -70
 #  define _NL_IDENTIFICATION_SOURCE       -71
 #  define _NL_IDENTIFICATION_ADDRESS      -72
@@ -291,11 +291,11 @@ typedef int nl_item;    /* Substitute 'int' for emulated nl_langinfo() */
 
 #  define HAS_MISSING_LANGINFO_ITEM_
 #endif
-#if defined(__ANDROID__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_MEASUREMENT)
+#if defined(__ANDROID__) || defined(__OHOS__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_MEASUREMENT)
 #  define _NL_MEASUREMENT_MEASUREMENT -85
 #  define HAS_MISSING_LANGINFO_ITEM_
 #endif
-#if defined(__ANDROID__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_NAME)
+#if defined(__ANDROID__) || defined(__OHOS__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_NAME)
 #  define _NL_NAME_NAME_FMT     -86
 #  define _NL_NAME_NAME_GEN     -87
 #  define _NL_NAME_NAME_MR      -88
@@ -305,13 +305,13 @@ typedef int nl_item;    /* Substitute 'int' for emulated nl_langinfo() */
 
 #  define HAS_MISSING_LANGINFO_ITEM_
 #endif
-#if defined(__ANDROID__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_PAPER)
+#if defined(__ANDROID__) || defined(__OHOS__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_PAPER)
 #  define _NL_PAPER_HEIGHT  -92
 #  define _NL_PAPER_WIDTH   -93
 
 #  define HAS_MISSING_LANGINFO_ITEM_
 #endif
-#if defined(__ANDROID__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_TELEPHONE)
+#if defined(__ANDROID__) || defined(__OHOS__) || ! defined(HAS_NL_LANGINFO) || ! defined(LC_TELEPHONE)
 #  define _NL_TELEPHONE_TEL_INT_FMT  -94
 #  define _NL_TELEPHONE_TEL_DOM_FMT  -95
 #  define _NL_TELEPHONE_INT_SELECT   -96
