diff --git a/src/runtime/napi/napi_body.rs b/src/runtime/napi/napi_body.rs
index b0717f39c6..5312407dcf 100644
--- a/src/runtime/napi/napi_body.rs
+++ b/src/runtime/napi/napi_body.rs
@@ -4040,7 +4040,8 @@ mod posix_platform_specific_v8_apis {
     not(windows),
     not(target_os = "android"),
     not(target_os = "macos"),
-    not(target_os = "freebsd")
+    not(target_os = "freebsd"),
+    not(target_env = "ohos")
 ))]
 mod posix_platform_specific_v8_apis {
     use core::ffi::c_void;
@@ -5246,7 +5247,8 @@ pub(crate) fn fix_dead_code_elimination() {
         not(windows),
         not(target_os = "android"),
         not(target_os = "macos"),
-        not(target_os = "freebsd")
+        not(target_os = "freebsd"),
+        not(target_env = "ohos")
     ))]
     keep_symbols!(
         posix_platform_specific_v8_apis::_ZN2v85Array3NewENS_5LocalINS_7ContextEEEmSt8functionIFNS_10MaybeLocalINS_5ValueEEEvEE,
