diff -ruN a/config.guess b/config.guess
--- a/config.guess	2025-12-18 19:39:40.000000000 +0800
+++ b/config.guess	2026-06-03 09:36:12.574706580 +0800
@@ -157,6 +157,8 @@
 	cat <<-EOF > "$dummy.c"
 	#if defined(__ANDROID__)
 	LIBC=android
+	#elif defined(__OHOS__)
+	LIBC=ohos
 	#else
 	#include <features.h>
 	#if defined(__UCLIBC__)
diff -ruN a/config.sub b/config.sub
--- a/config.sub	2025-12-18 19:39:40.000000000 +0800
+++ b/config.sub	2026-06-03 09:36:12.574706580 +0800
@@ -1768,7 +1768,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* | cos* | mbr* | ironclad* )
+            | fiwix* | mlibc* | cos* | mbr* | ironclad* | ohos* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
@@ -1829,7 +1829,7 @@
 case $kernel-$os-$obj in
 	linux-gnu*- | linux-android*- | linux-dietlibc*- | linux-llvm*- \
 		    | linux-mlibc*- | linux-musl*- | linux-newlib*- \
-		    | linux-relibc*- | linux-uclibc*- )
+                   | linux-relibc*- | linux-uclibc*- | linux-ohos*- )
 		;;
 	uclinux-uclibc*- )
 		;;
diff -ruN a/libraries/unix/config.guess b/libraries/unix/config.guess
--- a/libraries/unix/config.guess	2025-12-18 19:41:20.000000000 +0800
+++ b/libraries/unix/config.guess	2026-06-03 09:36:12.575706584 +0800
@@ -155,6 +155,9 @@
 
 	set_cc_for_build
 	cat <<-EOF > "$dummy.c"
+	#if defined(__OHOS__)
+	LIBC=ohos
+	#else
 	#include <features.h>
 	#if defined(__UCLIBC__)
 	LIBC=uclibc
diff -ruN a/libraries/unix/config.sub b/libraries/unix/config.sub
--- a/libraries/unix/config.sub	2025-12-18 19:41:20.000000000 +0800
+++ b/libraries/unix/config.sub	2026-06-03 09:36:12.575706584 +0800
@@ -1758,7 +1758,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* )
+            | fiwix* | ohos* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
@@ -1776,7 +1776,7 @@
 # (given a valid OS), if there is a kernel.
 case $kernel-$os in
 	linux-gnu* | linux-dietlibc* | linux-android* | linux-newlib* \
-		   | linux-musl* | linux-relibc* | linux-uclibc* )
+                  | linux-musl* | linux-relibc* | linux-uclibc* | linux-ohos* )
 		;;
 	uclinux-uclibc* )
 		;;
diff -ruN a/m4/ghc_convert_os.m4 b/m4/ghc_convert_os.m4
--- a/m4/ghc_convert_os.m4	2025-12-18 19:39:40.000000000 +0800
+++ b/m4/ghc_convert_os.m4	2026-06-03 09:36:12.575706584 +0800
@@ -10,6 +10,9 @@
       ios|watchos|tvos)
         $3="ios"
         ;;
+      ohos|linux-ohos*)
+        $3="linux"
+        ;;
       linux-android*)
         $3="linux-android"
         ;;
diff -ruN a/m4/ghc_llvm_target.m4 b/m4/ghc_llvm_target.m4
--- a/m4/ghc_llvm_target.m4	2025-12-18 19:39:40.000000000 +0800
+++ b/m4/ghc_llvm_target.m4	2026-06-03 09:36:12.576706588 +0800
@@ -36,7 +36,7 @@
       # for the LLVM Target. Otherwise these would be
       # turned into just `-linux` and fail to be found
       # in the `llvm-targets` file.
-      *-android*|*-gnueabi*|*-musleabi*)
+      *-android*|*-gnueabi*|*-musleabi*|*-ohos*)
         GHC_CONVERT_VENDOR([$3],[llvm_target_vendor])
         llvm_target_os="$4"
         ;;
diff -ruN a/utils/ghc-toolchain/src/GHC/Toolchain/ParseTriple.hs b/utils/ghc-toolchain/src/GHC/Toolchain/ParseTriple.hs
--- a/utils/ghc-toolchain/src/GHC/Toolchain/ParseTriple.hs	2025-12-18 19:39:41.000000000 +0800
+++ b/utils/ghc-toolchain/src/GHC/Toolchain/ParseTriple.hs	2026-06-03 09:36:12.576706588 +0800
@@ -63,6 +63,8 @@
     case os of
       "linux" -> pure OSLinux
       "linux-android" -> pure OSLinux
+      "ohos" -> pure OSLinux
+      "linux-ohos" -> pure OSLinux
       "darwin" -> pure OSDarwin
       "ios" -> pure OSDarwin
       "watchos" -> pure OSDarwin
