--- a/configure.ac
+++ b/configure.ac
@@ -24,7 +24,7 @@
 AC_CHECK_LIB([wolfssl], [wc_Chacha_Process],
   [add_cryptcab_support=yes],
   [add_cryptcab_support=no ; warn_cryptcab=yes])
-AC_CHECK_LIB([pthread], [pthread_create],
+AC_CHECK_LIB([pthread], [pthread_cancel],
   [enable_router=yes],
   [enable_router=no ; warn_router=yes])
 AC_CHECK_LIB([pcap], [pcap_open_dead],
--- a/src/vdetaplib/libvdetap.c
+++ b/src/vdetaplib/libvdetap.c
@@ -42,7 +42,9 @@
 
 static int     (*native_ioctl) (int d, int request, ...) = NULL;
 static int     (*native_open) (const char *pathname, int flags, ...) = NULL;
+#ifndef open64
 static int     (*native_open64) (const char *pathname, int flags, ...) = NULL;
+#endif
 
 int tapfd[2] = {-1,-1};
 static int tapcount=0;
@@ -87,7 +89,9 @@
 	int i;
 	nativesym(ioctl, "ioctl");
 	nativesym(open, "open");
+#ifndef open64
 	nativesym(open64, "open64");
+#endif
 	for (i=1;i<MAX;i++) 
 		pidpool[i-1].next= &(pidpool[i]);
 	flh=pidpool;
@@ -121,6 +125,7 @@
 		return native_open(path, flags, data);
 }
 
+#ifndef open64
 int open64(const char *path, int flags, ...)
 {
 	va_list ap;
@@ -140,6 +145,8 @@
 		return native_open64(path, flags | O_LARGEFILE, data);
 }
 
+#endif
+
 static char *getvdeopt(struct ifreq *ifr,char *suffix)
 {
 	static char buf[32];
