diff --git a/src/runtime/shell/builtin/echo.rs b/src/runtime/shell/builtin/echo.rs
index ee1163375..858f0413d 100644
--- a/src/runtime/shell/builtin/echo.rs
+++ b/src/runtime/shell/builtin/echo.rs
@@ -113,7 +113,7 @@ impl Echo {
         Builtin::done(
             interp,
             cmd,
-            err.map(|e| e.errno as crate::shell::ExitCode).unwrap_or(0),
+            err.map(|e| e.errno.unsigned_abs() as crate::shell::ExitCode).unwrap_or(0),
         )
     }
 }
