diff --git a/src/runtime/shell/builtin/which.rs b/src/runtime/shell/builtin/which.rs
index 0a39513800..b03a196f98 100644
--- a/src/runtime/shell/builtin/which.rs
+++ b/src/runtime/shell/builtin/which.rs
@@ -181,7 +181,7 @@ impl Which {
         e: Option<bun_sys::SystemError>,
     ) -> Yield {
         if let Some(err) = e {
-            return Builtin::done(interp, cmd, err.errno as crate::shell::ExitCode);
+            return Builtin::done(interp, cmd, err.errno.unsigned_abs() as crate::shell::ExitCode);
         }
         match Self::state_mut(interp, cmd).state {
             State::OneArg => Builtin::done(interp, cmd, 1),
