diff --git a/src/unix.rs b/src/unix.rs
index 820d897..97964d1 100644
--- a/src/unix.rs
+++ b/src/unix.rs
@@ -73,7 +73,7 @@ extern {
     fn errno_location() -> *mut c_int;
 
     #[cfg(feature = "std")]
-    #[cfg_attr(target_os = "linux", link_name = "__xpg_strerror_r")]
+    #[cfg_attr(all(target_os = "linux", not(any(target_env = "musl", target_env = "ohos"))), link_name = "__xpg_strerror_r")]
     fn strerror_r(errnum: c_int, buf: *mut c_char,
                   buflen: libc::size_t) -> c_int;
 }
