diff -ruN openssl-4.0.2/crypto/armcap.c openssl-4.0.2-new/crypto/armcap.c
--- openssl-4.0.2/crypto/armcap.c	2026-08-25 11:38:35.000000000 +0000
+++ openssl-4.0.2-new/crypto/armcap.c	2026-08-31 09:37:23.571448731 +0000
@@ -460,6 +460,17 @@
             || MIDR_IS_CPU_MODEL(OPENSSL_arm_midr, ARM_CPU_IMP_NVIDIA, NVIDIA_CPU_PART_OLYMPUS)))
         OPENSSL_armcap_P |= ARMV8_HAVE_SHA3_AND_WORTH_USING;
 
+#ifdef __OHOS__
+    /*
+     * Some OpenHarmony devices use heterogeneous cores where
+     * only part of the cores support SVE2. If OpenSSL probes CPU features
+     * on an SVE2-capable core but is then scheduled onto a core without
+     * SVE2 when the SVE2 assembly is executed, SIGILL is raised. Disable
+     * the SVE2 path.
+     */
+    OPENSSL_armcap_P &= ~ARMV9_SVE2;
+#endif
+
     if (OPENSSL_armcap_P & ARMV9_SVE2) {
         uint64_t vl_bytes = _armv8_sve_get_vl_bytes();
 
