diff --git a/src/runtime/cli/upgrade_command.rs b/src/runtime/cli/upgrade_command.rs
index 8ceab6b44f..14f6510ff8 100644
--- a/src/runtime/cli/upgrade_command.rs
+++ b/src/runtime/cli/upgrade_command.rs
@@ -105,10 +105,12 @@ impl Version {
     };
     pub(crate) const TRIPLET: &'static str =
         const_format::concatcp!(Version::PLATFORM_LABEL, "-", Version::ARCH_LABEL);
-    const SUFFIX_ABI: &'static str = if Environment::IS_MUSL {
-        "-musl"
+    const SUFFIX_ABI: &'static str = if Environment::IS_OHOS {
+        "-ohos"
     } else if Environment::IS_ANDROID {
         "-android"
+    } else if Environment::IS_MUSL {
+        "-musl"
     } else {
         ""
     };
