--- a/hw/arm/fsl-imx8mm.c	2026-10-03 06:38:48.514442500 +0800
+++ b/hw/arm/fsl-imx8mm.c	2026-10-03 06:38:48.606443438 +0800
@@ -370,7 +370,7 @@
 
     /* UARTs */
     for (i = 0; i < FSL_IMX8MM_NUM_UARTS; i++) {
-        static const struct {
+        const struct {
             hwaddr addr;
             unsigned int irq;
         } serial_table[FSL_IMX8MM_NUM_UARTS] = {
@@ -448,7 +448,7 @@
 
     /* I2Cs */
     for (i = 0; i < FSL_IMX8MM_NUM_I2CS; i++) {
-        static const struct {
+        const struct {
             hwaddr addr;
             unsigned int irq;
         } i2c_table[FSL_IMX8MM_NUM_I2CS] = {
@@ -469,7 +469,7 @@
 
     /* GPIOs */
     for (i = 0; i < FSL_IMX8MM_NUM_GPIOS; i++) {
-        static const struct {
+        const struct {
             hwaddr addr;
             unsigned int irq_low;
             unsigned int irq_high;
@@ -517,7 +517,7 @@
 
     /* USDHCs */
     for (i = 0; i < FSL_IMX8MM_NUM_USDHCS; i++) {
-        static const struct {
+        const struct {
             hwaddr addr;
             unsigned int irq;
         } usdhc_table[FSL_IMX8MM_NUM_USDHCS] = {
@@ -537,7 +537,7 @@
 
     /* USBs */
     for (i = 0; i < FSL_IMX8MM_NUM_USBS; i++) {
-        static const struct {
+        const struct {
             hwaddr addr;
             unsigned int irq;
         } usb_table[FSL_IMX8MM_NUM_USBS] = {
@@ -558,7 +558,7 @@
 
     /* ECSPIs */
     for (i = 0; i < FSL_IMX8MM_NUM_ECSPIS; i++) {
-        static const struct {
+        const struct {
             hwaddr addr;
             unsigned int irq;
         } spi_table[FSL_IMX8MM_NUM_ECSPIS] = {
@@ -600,7 +600,7 @@
 
     /* Watchdogs */
     for (i = 0; i < FSL_IMX8MM_NUM_WDTS; i++) {
-        static const struct {
+        const struct {
             hwaddr addr;
             unsigned int irq;
         } wdog_table[FSL_IMX8MM_NUM_WDTS] = {
--- a/hw/arm/fsl-imx8mp.c	2026-10-03 06:38:48.514442500 +0800
+++ b/hw/arm/fsl-imx8mp.c	2026-10-03 06:38:48.606443438 +0800
@@ -676,7 +676,7 @@
 
     /* FlexCANs */
     for (i = 0; i < FSL_IMX8MP_NUM_CANS; i++) {
-        static const struct {
+        const struct {
             hwaddr addr;
             unsigned int irq;
         } flexcan_table[FSL_IMX8MP_NUM_CANS] = {
