--- a/linux-user/mmap.c	2026-09-29 21:33:21.000000000 +0800
+++ b/linux-user/mmap.c	2026-10-03 06:44:59.702443438 +0800
@@ -1345,8 +1345,12 @@
     case MADV_UNMERGEABLE:
     case MADV_HUGEPAGE:
     case MADV_NOHUGEPAGE:
+#ifdef MADV_POPULATE_READ
     case MADV_POPULATE_READ:
+#endif
+#ifdef MADV_POPULATE_WRITE
     case MADV_POPULATE_WRITE:
+#endif
 #ifdef MADV_COLLAPSE
     case MADV_COLLAPSE:
 #endif
