--- a/linux-user/syscall.c	2026-10-03 18:25:34.235689532 +0800
+++ b/linux-user/syscall.c	2026-10-03 18:25:34.235689532 +0800
@@ -13684,7 +13684,12 @@
             if (!p) {
                 return -TARGET_EFAULT;
             }
+#ifdef __OHOS__
+            ret = get_errno(syscall(__NR_mq_open, (char *)p + (*(char *)p == '/'), host_flags,
+                                    arg3, pposix_mq_attr));
+#else
             ret = get_errno(mq_open(p, host_flags, arg3, pposix_mq_attr));
+#endif
             unlock_user (p, arg1, 0);
         }
         return ret;
@@ -13694,7 +13699,11 @@
         if (!p) {
             return -TARGET_EFAULT;
         }
+#ifdef __OHOS__
+        ret = get_errno(syscall(__NR_mq_unlink, (char *)p + (*(char *)p == '/')));
+#else
         ret = get_errno(mq_unlink(p));
+#endif
         unlock_user (p, arg1, 0);
         return ret;
 
@@ -13805,10 +13814,20 @@
             ret = 0;
             if (arg2 != 0) {
                 copy_from_user_mq_attr(&posix_mq_attr_in, arg2);
+#ifdef __OHOS__
+                ret = get_errno(syscall(__NR_mq_getsetattr, arg1,
+                                        &posix_mq_attr_in, &posix_mq_attr_out));
+#else
                 ret = get_errno(mq_setattr(arg1, &posix_mq_attr_in,
                                            &posix_mq_attr_out));
+#endif
             } else if (arg3 != 0) {
+#ifdef __OHOS__
+                ret = get_errno(syscall(__NR_mq_getsetattr, arg1, NULL,
+                                        &posix_mq_attr_out));
+#else
                 ret = get_errno(mq_getattr(arg1, &posix_mq_attr_out));
+#endif
             }
             if (ret == 0 && arg3 != 0) {
                 copy_to_user_mq_attr(arg3, &posix_mq_attr_out);
