diff --git a/libs/mysql/test/Jamfile b/libs/mysql/test/Jamfile
index 16c66f9d..276109f9 100644
--- a/libs/mysql/test/Jamfile
+++ b/libs/mysql/test/Jamfile
@@ -76,7 +76,7 @@ local requirements =
         # gcc-11 emits spurious warnings for valid vector::insert ops
         <toolset>gcc-11:<cxxflags>-Wno-stringop-overflow
         # gcc-15 emits spurious warnings about uninitialized variables in coroutines
-        <toolset>gcc-15:<cxxflags>-Wno-maybe-uninitialized
+        <toolset>gcc-15:<cxxflags>-Wno-uninitialized
         # TODO: remove when PFR unused warnings are fixed
         # https://github.com/boostorg/pfr/pull/187
         <toolset>gcc:<cxxflags>-Wno-unused-parameter
diff --git a/libs/spirit/build.jam b/libs/spirit/build.jam
index 0ef57f79..f1a321a7 100644
--- a/libs/spirit/build.jam
+++ b/libs/spirit/build.jam
@@ -62,7 +62,7 @@ project /boost/spirit
         <warnings>extra,<toolset>msvc:<cxxflags>-w44555  # result of expression not used
 
         # Broken or very high false-positive rate
-        <known-warnings>hide,<toolset>gcc:<cxxflags>-Wno-maybe-uninitialized  # Too many pesky false-positives
+        <known-warnings>hide,<toolset>gcc:<cxxflags>-Wno-uninitialized  # Too many pesky false-positives
         <known-warnings>hide,<toolset>msvc-$(9.0-14.0):<cxxflags>-wd4100  # unreferenced formal parameter
         <known-warnings>hide,<toolset>msvc-$(9.0-14.0):<cxxflags>-wd4512  # assignment operator could not be generated
         <known-warnings>hide,<toolset>msvc-$(9.0-14.0):<cxxflags>-wd4714  # function 'x' marked as __forceinline not inlined
diff --git a/libs/url/build/Jamfile b/libs/url/build/Jamfile
index bd4748fc..fe83d503 100644
--- a/libs/url/build/Jamfile
+++ b/libs/url/build/Jamfile
@@ -33,7 +33,7 @@ project
       <define>BOOST_URL_SOURCE
       <toolset>msvc-14.0:<build>no
       # Warnings in dependencies
-      <toolset>gcc:<cxxflags>"-Wno-maybe-uninitialized"
+      <toolset>gcc:<cxxflags>"-Wno-uninitialized"
       <toolset>gcc,<target-os>windows:<cxxflags>"-Wno-error=array-bounds"
     : common-requirements
       <link>shared:<define>BOOST_URL_DYN_LINK=1
