diff --git a/libs/locale/src/posix/numeric.cpp b/libs/locale/src/posix/numeric.cpp
index 2810b821..8ce50633 100644
--- a/libs/locale/src/posix/numeric.cpp
+++ b/libs/locale/src/posix/numeric.cpp
@@ -29,6 +29,7 @@
 
 #include "../util/numeric.hpp"
 #include "all_generator.hpp"
+#include "../../../musl/strfmon.hpp"
 
 namespace boost { namespace locale { namespace impl_posix {
 
diff --git a/libs/locale/test/test_posix_formatting.cpp b/libs/locale/test/test_posix_formatting.cpp
index 840eac89..b578f4c5 100644
--- a/libs/locale/test/test_posix_formatting.cpp
+++ b/libs/locale/test/test_posix_formatting.cpp
@@ -16,6 +16,7 @@
 #ifndef BOOST_LOCALE_NO_POSIX_BACKEND
 #    include <langinfo.h>
 #    include <monetary.h>
+#    include "../../musl/strfmon.hpp"
 #endif
 #include "boostLocale/test/tools.hpp"
 #include "boostLocale/test/unit_test.hpp"
diff --git a/libs/musl/strfmon.hpp b/libs/musl/strfmon.hpp
new file mode 100644
index 00000000..0c12c236
--- /dev/null
+++ b/libs/musl/strfmon.hpp
@@ -0,0 +1,94 @@
+#include <stdio.h>
+#include <ctype.h>
+#include <stdarg.h>
+#include <monetary.h>
+#include <errno.h>
+
+extern "C" {
+
+static ssize_t vstrfmon_l(char *s, size_t n, locale_t loc, const char *fmt, va_list ap)
+{
+	size_t l;
+	double x;
+	int fill, nogrp, negpar, nosym, left, intl;
+	int lp, rp, w, fw;
+	char *s0=s;
+	for (; n && *fmt; ) {
+		if (*fmt != '%') {
+		literal:
+			*s++ = *fmt++;
+			n--;
+			continue;
+		}
+		fmt++;
+		if (*fmt == '%') goto literal;
+
+		fill = ' ';
+		nogrp = 0;
+		negpar = 0;
+		nosym = 0;
+		left = 0;
+		for (; ; fmt++) {
+			switch (*fmt) {
+			case '=':
+				fill = *++fmt;
+				continue;
+			case '^':
+				nogrp = 1;
+				continue;
+			case '(':
+				negpar = 1;
+			case '+':
+				continue;
+			case '!':
+				nosym = 1;
+				continue;
+			case '-':
+				left = 1;
+				continue;
+			}
+			break;
+		}
+
+		for (fw=0; isdigit(*fmt); fmt++)
+			fw = 10*fw + (*fmt-'0');
+		lp = 0;
+		rp = 2;
+		if (*fmt=='#') for (lp=0, fmt++; isdigit(*fmt); fmt++)
+			lp = 10*lp + (*fmt-'0');
+		if (*fmt=='.') for (rp=0, fmt++; isdigit(*fmt); fmt++)
+			rp = 10*rp + (*fmt-'0');
+
+		intl = *fmt++ == 'i';
+
+		w = lp + 1 + rp;
+		if (!left && fw>w) w = fw;
+
+		x = va_arg(ap, double);
+		l = snprintf(s, n, "%*.*f", w, rp, x);
+		if (l >= n) {
+			errno = E2BIG;
+			return -1;
+		}
+		s += l;
+		n -= l;
+
+		/* fix warning: variable set but not used */
+		(void)fill; (void)nogrp; (void)negpar; (void)nosym; (void)left; (void)intl;
+	}
+	return s-s0;
+}
+
+static ssize_t strfmon_l(char *__restrict__ s, size_t n, locale_t loc, const char *__restrict__ fmt, ...)
+{
+	va_list ap;
+	ssize_t ret;
+
+	va_start(ap, fmt);
+	ret = vstrfmon_l(s, n, loc, fmt, ap);
+	va_end(ap);
+
+	return ret;
+}
+
+}
