diff --git a/src/jsc/bindings/BunProcess.cpp b/src/jsc/bindings/BunProcess.cpp
index 86a9622c33..51e03c9ca7 100644
--- a/src/jsc/bindings/BunProcess.cpp
+++ b/src/jsc/bindings/BunProcess.cpp
@@ -192,6 +192,8 @@ static JSValue constructPlatform(VM& vm, JSObject* processObject)
 {
 #if defined(__APPLE__)
     return JSC::jsString(vm, makeAtomString("darwin"_s));
+#elif defined(__OHOS__)
+    return JSC::jsString(vm, makeAtomString("openharmony"_s));
 #elif defined(__ANDROID__)
     return JSC::jsString(vm, makeAtomString("android"_s));
 #elif defined(__linux__)
