diff --git a/scripts/build/codegen.ts b/scripts/build/codegen.ts
index 83c5c0f91e..73772c78ce 100644
--- a/scripts/build/codegen.ts
+++ b/scripts/build/codegen.ts
@@ -100,15 +100,17 @@ function readPackageDeps(pkgDir: string): string[] {
  */
 function codegenTarget(cfg: Config): { platform: string; arch: string } {
   const platform =
-    cfg.abi === "android"
-      ? "android"
-      : cfg.os === "darwin"
-        ? "darwin"
-        : cfg.os === "windows"
-          ? "win32"
-          : cfg.os === "freebsd"
-            ? "freebsd"
-            : "linux";
+    cfg.ohos
+      ? "openharmony"
+      : cfg.abi === "android"
+        ? "android"
+        : cfg.os === "darwin"
+          ? "darwin"
+          : cfg.os === "windows"
+            ? "win32"
+            : cfg.os === "freebsd"
+              ? "freebsd"
+              : "linux";
   const arch = cfg.x64 ? "x64" : "arm64";
   return { platform, arch };
 }
