diff --git a/src/bun_core/env.rs b/src/bun_core/env.rs
index cbf6473fc2..6ecf4f4a0d 100644
--- a/src/bun_core/env.rs
+++ b/src/bun_core/env.rs
@@ -36,8 +36,10 @@ pub(crate) const IS_FREEBSD: bool = cfg!(target_os = "freebsd");
 pub const IS_KQUEUE: bool = IS_MAC || IS_FREEBSD;
 pub(crate) const IS_AARCH64: bool = cfg!(target_arch = "aarch64");
 pub(crate) const IS_X64: bool = cfg!(target_arch = "x86_64");
-pub const IS_MUSL: bool = cfg!(target_env = "musl");
+pub const IS_MUSL: bool = cfg!(any(target_env = "musl", target_env = "ohos"));
 pub const IS_ANDROID: bool = cfg!(target_os = "android");
+pub const IS_OHOS: bool = cfg!(target_env = "ohos");
+pub const IS_GLIBC: bool = IS_LINUX && cfg!(target_env = "gnu");
 pub const ALLOW_ASSERT: bool = IS_DEBUG || IS_TEST || build_options::RELEASE_SAFE;
 pub const CI_ASSERT: bool =
     IS_DEBUG || IS_TEST || ENABLE_ASAN || (build_options::RELEASE_SAFE && IS_CANARY);
