diff --git a/src/js/node/os.ts b/src/js/node/os.ts
index 9995113e33..e4010d1ea7 100644
--- a/src/js/node/os.ts
+++ b/src/js/node/os.ts
@@ -121,7 +121,7 @@ function bound(binding) {
         ? "Windows_NT"
         : process.platform === "darwin"
           ? "Darwin"
-          : process.platform === "linux" || process.platform === "android"
+          : process.platform === "linux" || process.platform === "android" || process.platform === "openharmony"
             ? "Linux"
             : process.platform === "freebsd"
               ? "FreeBSD"
@@ -135,7 +135,7 @@ function bound(binding) {
       // separate PR to avoid behavior change in the Android port.
       // FreeBSD: uname -m returns MACHINE ("arm64"/"amd64"), not MACHINE_ARCH.
       return process.arch === "arm64"
-        ? process.platform === "android"
+        ? process.platform === "android" || process.platform === "openharmony"
           ? "aarch64"
           : "arm64"
         : process.arch === "x64"
