diff --git a/test/js/node/os/os.test.js b/test/js/node/os/os.test.js
--- a/test/js/node/os/os.test.js
+++ b/test/js/node/os/os.test.js
@@ -76,3 +76,11 @@ it("tmpdir", () => {
     const originalEnv = process.env.TMPDIR;
-    let dir = process.env.TMPDIR || process.env.TMP || process.env.TEMP || "/tmp";
+    let dir =
+      process.env.TMPDIR ||
+      process.env.TMP ||
+      process.env.TEMP ||
+      (process.platform === "android"
+        ? "/data/local/tmp"
+        : process.platform === "openharmony"
+          ? "/data/storage/el2/base/cache"
+          : "/tmp");
     if (dir.length > 1 && dir.endsWith("/")) {
