diff --git a/hostfile.c b/hostfile.c
index 033b291..6c6ef67 100644
--- a/hostfile.c
+++ b/hostfile.c
@@ -729,7 +729,11 @@ hostfile_replace_entries(const char *filename, const char *host, const char *ip,
 			r = SSH_ERR_SYSTEM_ERROR;
 			goto fail;
 		}
+#ifdef __OHOS__
+		if (rename(filename, back) == -1) {
+#else
 		if (link(filename, back) == -1) {
+#endif
 			oerrno = errno;
 			error_f("link %.100s to %.100s: %s", filename,
 			    back, strerror(errno));
