diff --git a/openbsd-compat/xcrypt.c b/openbsd-compat/xcrypt.c
index 9cded66..1736209 100644
--- a/openbsd-compat/xcrypt.c
+++ b/openbsd-compat/xcrypt.c
@@ -70,14 +70,15 @@
 static const char *
 pick_salt(void)
 {
-	struct passwd *pw;
-	char *passwd, *p;
-	size_t typelen;
 	static char salt[32];
 
 	if (salt[0] != '\0')
 		return salt;
 	strlcpy(salt, "xx", sizeof(salt));
+#ifndef __OHOS__
+	struct passwd *pw;
+	char *passwd, *p;
+	size_t typelen;
 	setpwent();
 	while ((pw = getpwent()) != NULL) {
 		if ((passwd = shadow_pw(pw)) == NULL)
@@ -91,6 +92,7 @@ pick_salt(void)
 	}
  out:
 	endpwent();
+#endif
 	return salt;
 }
 
@@ -130,7 +132,7 @@ shadow_pw(struct passwd *pw)
 {
 	char *pw_password = pw->pw_passwd;
 
-# if defined(HAVE_SHADOW_H) && !defined(DISABLE_SHADOW)
+# if !defined(__OHOS__) && defined(HAVE_SHADOW_H) && !defined(DISABLE_SHADOW)
 	struct spwd *spw = getspnam(pw->pw_name);
 
 	if (spw != NULL)
