diff --git a/ssh-agent.c b/ssh-agent.c
index 5fc73d6..e3e1d4b 100644
--- a/ssh-agent.c
+++ b/ssh-agent.c
@@ -83,7 +83,7 @@
 #include "version.h"
 
 #ifndef DEFAULT_ALLOWED_PROVIDERS
-# define DEFAULT_ALLOWED_PROVIDERS "/usr/lib*/*,/usr/local/lib*/*"
+# define DEFAULT_ALLOWED_PROVIDERS "@@HOMEBREW_PREFIX@@/lib*/*,@@HOMEBREW_PREFIX@@/lib*/*"
 #endif
 #ifndef DEFAULT_WEBSAFE_ALLOWLIST
 # define DEFAULT_WEBSAFE_ALLOWLIST "ssh:*"
@@ -2265,8 +2265,10 @@ main(int ac, char **av)
 	sanitise_stdfd();
 
 	/* drop */
+#ifndef __OHOS__
 	(void)setegid(getgid());
 	(void)setgid(getgid());
+#endif
 
 	platform_disable_tracing(0);	/* strict=no */
 
