diff --git a/sshd.c b/sshd.c
index 7d1466d..e2b5c8a 100644
--- a/sshd.c
+++ b/sshd.c
@@ -1317,7 +1317,12 @@ main(int ac, char **av)
 	saved_argc = ac;
 	rexec_argc = ac;
 	saved_argv = xcalloc(ac + 1, sizeof(*saved_argv));
+#ifdef __OHOS__
+	saved_argv[0] = xstrdup("@@HOMEBREW_PREFIX@@/sbin/sshd");
+	for (i = 1; (int)i < ac; i++)
+#else
 	for (i = 0; (int)i < ac; i++)
+#endif
 		saved_argv[i] = xstrdup(av[i]);
 	saved_argv[i] = NULL;
 
@@ -1327,8 +1332,10 @@ main(int ac, char **av)
 	av = saved_argv;
 #endif
 
+#ifndef __OHOS__
 	if (geteuid() == 0 && setgroups(0, NULL) == -1)
 		debug("setgroups(): %.200s", strerror(errno));
+#endif
 
 	/* Ensure that fds 0, 1 and 2 are open or directed to /dev/null */
 	sanitise_stdfd();
@@ -1451,8 +1458,10 @@ main(int ac, char **av)
 			break;
 		}
 	}
+#ifndef __OHOS__
 	if (!test_flag && !inetd_flag && !do_dump_cfg && !path_absolute(av[0]))
 		fatal("sshd requires execution with an absolute path");
+#endif
 
 	closefrom(STDERR_FILENO + 1);
 
@@ -1720,12 +1729,15 @@ main(int ac, char **av)
 	/* Ensure privsep directory is correctly configured. */
 	need_chroot = ((getuid() == 0 || geteuid() == 0) ||
 	    options.kerberos_authentication);
+#ifndef __OHOS__
 	if ((getpwnam(SSH_PRIVSEP_USER)) == NULL && need_chroot) {
 		fatal("Privilege separation user %s does not exist",
 		    SSH_PRIVSEP_USER);
 	}
 	endpwent();
+#endif
 
+#ifndef __OHOS__
 	if (need_chroot) {
 		if ((stat(_PATH_PRIVSEP_CHROOT_DIR, &sb) == -1) ||
 		    (S_ISDIR(sb.st_mode) == 0))
@@ -1741,6 +1753,7 @@ main(int ac, char **av)
 			fatal("%s must be owned by root and not group or "
 			    "world-writable.", _PATH_PRIVSEP_CHROOT_DIR);
 	}
+#endif
 
 	if (test_flag > 1)
 		print_config(&connection_info);
@@ -1762,8 +1775,10 @@ main(int ac, char **av)
 	 * to create a file, and we can't control the code in every
 	 * module which might be used).
 	 */
+#ifndef __OHOS__
 	if (setgroups(0, NULL) < 0)
 		debug("setgroups() failed: %.200s", strerror(errno));
+#endif
 
 	/* Prepare arguments for sshd-session */
 	if (rexec_argc < 0)
@@ -1810,7 +1825,7 @@ main(int ac, char **av)
 	already_daemon = daemonized();
 	if (!(debug_flag || inetd_flag || no_daemon_flag || already_daemon)) {
 
-		if (daemon(0, 0) == -1)
+		if (daemon(1, 0) == -1)
 			fatal("daemon() failed: %.200s", strerror(errno));
 
 		disconnect_controlling_tty();
@@ -1822,8 +1837,10 @@ main(int ac, char **av)
 	 * Chdir to the root directory so that the current disk can be
 	 * unmounted if desired.
 	 */
+#ifndef __OHOS__
 	if (chdir("/") == -1)
 		error("chdir(\"/\"): %s", strerror(errno));
+#endif
 
 	/* ignore SIGPIPE */
 	ssh_signal(SIGPIPE, SIG_IGN);
