diff --git a/src/runtime/napi/napi_body.rs b/src/runtime/napi/napi_body.rs
index 600d9dcfac..1590833a52 100644
--- a/src/runtime/napi/napi_body.rs
+++ b/src/runtime/napi/napi_body.rs
@@ -3253,7 +3253,8 @@ mod posix_platform_specific_v8_apis {
     not(windows),
     not(target_os = "android"),
     not(target_os = "macos"),
-    not(target_os = "freebsd")
+    not(target_os = "freebsd"),
+    not(target_env = "ohos")
 ))]
 mod posix_platform_specific_v8_apis {
     use core::ffi::c_void;
@@ -4223,7 +4224,8 @@ pub fn fix_dead_code_elimination() {
         not(windows),
         not(target_os = "android"),
         not(target_os = "macos"),
-        not(target_os = "freebsd")
+        not(target_os = "freebsd"),
+        not(target_env = "ohos")
     ))]
     keep_symbols!(posix_platform_specific_v8_apis::_ZN2v85Array3NewENS_5LocalINS_7ContextEEEmSt8functionIFNS_10MaybeLocalINS_5ValueEEEvEE);
 
